src/routes/auth/login.test.ts

```diff
@@ -0,0 +1,161 @@
+import { login } from './login'
+import jws from 'jws'
+import { createSpyObj } from '../../test/helper/jest'
+import { Environment } from '../../utils/Environment'
+import * as val from 'express-validator'
+jest.mock('express-validator')
+
+describe('Check login', () => {
+  let resSpy
+  let req
+
+  beforeEach(() => {
+    resSpy = createSpyObj('Response', ['status', 'json', 'end', 'send'])
+    req = {
+      body: {
+        username: 'admin',
+        password: 'Passw0rd'
+      }
+    }
+    resSpy.status.mockReturnThis()
+    resSpy.json.mockReturnThis()
+    resSpy.send.mockReturnThis()
+    jest.spyOn(val, 'validationResult').mockImplementation(() => {
+      return {
+        isEmpty: jest.fn().mockReturnValue(true),
+        array: jest.fn().mockReturnValue([{ test: 'error' }])
+      } as any
+    })
+  })
+  it('should pass with correct user and password', async () => {
+    Environment.Config = {
```

Comment:
i think define the Environment.Config in beforeEach -- since that is reused across all of these tests, and in the test just update the single value of concern (i.e. web_admin_user). This will reduce duplicate code.

Reply:
Done